### PR TITLE
434 step goal command

### PR DESCRIPTION
**Pull request by bscheltinga**
Fixes: https://github.com/PerfectFit-project/virtual-coach-issues/issues/434

@wbaccinelli note, I did not test this yet.

### CODE REVIEW

File: Rasa_Bot/domain/domain_idle_commands.yml [hunk 1]
```diff
@@ -23,11 +31,18 @@ responses:
     - text: "Excuses, er is geen onvolledige conversatie om af te ronden."
     - text: "Het spijt me, er is geen onafgemaakte dialoog om te voltooien."
 
+  utter_idle_commands_today_steps:
+    - text: "Tot nu toe heb je {idle_commands_today_steps} stappen gezet vandaag. 🚶🏃"
+
+  utter_idle_commands_today_goal:
+    - text: "Het stappendoel voor vandaag is {pa_goal} stappen. 🎯"
```
Comment (wbaccinelli) [inline Rasa_Bot/domain/domain_idle_commands.yml:38]: I cannot find where this slot is defined and set. Are you sure that it is managed somewhere?

**Review by wbaccinelli — CHANGES_REQUESTED** [review]
Good to have the new functionality! I tested it but the intent is not recognized and I got some errors due to the empty steps and the unset slot. I left few comments. can you address them?

File: Rasa_Bot/actions/actions_idle_commands.py [hunk 2]
```diff
@@ -148,3 +157,22 @@ async def run(self, dispatcher, tracker, domain):
...
+        # get smoking status with 1: not (re)lapsed and 2: did (re)lapse last 4 weeks
+
+        user_id = int(tracker.current_state()['sender_id'])
+
+        start = datetime.datetime.now()
+        end = start + datetime.timedelta(days=1)
+
+        # get number of steps from today
+        steps_data = get_steps_data(user_id=user_id, start_date=start, end_date=end)
+        today_steps = steps_data[0]['steps']
```
[the remark this answers is not in the document]
Comment (bscheltinga) [reply Rasa_Bot/actions/actions_idle_commands.py:176]: Yes, that could occur. Nice catch

File: Rasa_Bot/domain/domain_idle_commands.yml (hunk 1 above)
Comment (bscheltinga) [reply Rasa_Bot/domain/domain_idle_commands.yml:38]: Good catch, I looked at another action (action_notifications_get_daily_step_goal) and took not the slot name, but variable name. Fixed it!

File: Rasa_Bot/actions/actions_idle_commands.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (bscheltinga) [reply Rasa_Bot/actions/actions_idle_commands.py:176]: Fixed this in other functions as well

**Comment by bscheltinga** [comment]
> Good to have the new functionality! I tested it but the intent is not recognized and I got some errors due to the empty steps and the unset slot. I left few comments. can you address them?

I tried the intent last time, but it indeed was not working. I added an intent typo-option, retrained and now the intent is recognized :)

**Comment by bscheltinga** [comment]
> > Good to have the new functionality! I tested it but the intent is not recognized and I got some errors due to the empty steps and the unset slot. I left few comments. can you address them?
> 
> I tried the intent last time, but it indeed was not working. I added an intent typo-option, retrained and now the intent is recognized :)

But now the 'uitleg' intent is not recognized...

**Review by wbaccinelli — APPROVED** [review]
Great, it works for me now!

**Comment by bscheltinga** [comment]
I did the merge, feels a bit like my 'last dance'...

**Pull request merged**